core/modules/replica/ServiceProvider.h

```diff
@@ -132,6 +133,13 @@ class ServiceProvider : public std::enable_shared_from_this<ServiceProvider> {
     /// @return a reference to worker messenger service (configured for controllers)
     MessengerPtr const& messenger();
 
+    /**
+     * @param database The name of a named mutex.
+     * @return A smart pointer to a mutex for the name.
+     * @throw std::invalid_argument If the name is empty.
```

Comment:
I'd add a comment to this that the shared pointer must live at least as long as the mutex is locked, or bad things could happen.

Reply:
Good suggestion!